bt-115: Create sign in sign up pages

JosuerBague opened the pull request:
- Created Sign Up Form UI
- Created Sign In Form UI

TODO:
Connecting forms to API

JosuerBague committed:
bt-115: + Create reusable google auth button

JosuerBague committed:
bt-115: + Add variables

JosuerBague committed:
bt-115: + Add UserSignInValidationSchema

JosuerBague committed:
bt-115: + Add UserSignInRequestDto

JosuerBague committed:
bt-115: + Update exports from shared dir

JosuerBague committed:
bt-115: + Add classNames props to input and link component

JosuerBague committed:
bt-115: + Add ReactNode type for checkbox label prop

JosuerBague committed:
bt-115: + Create sign-in form UI

JosuerBague committed:
Merge branch 'development' into task/bt-115-create-sign-in-sign-up-pages

JosuerBague committed:
bt-115: * Add classNames to input and link components

JosuerBague committed:
bt-115: + Createdd sign-in layout

JosuerBague committed:
bt-115: + Create sign-up form

JosuerBague committed:
Merge branch 'development' into task/bt-115-create-sign-in-sign-up-pages

JosuerBague committed:
bt-115: * Minor css fix

JosuerBague commented:
Sign Up Talent:
![image](https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/assets/90120572/903b10ed-67a8-4c8a-8afa-caff255167be)

Sign In Talent:
![image](https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/assets/90120572/bef1c137-fee3-4a2c-b2f5-7b33f5875804)

Sign Up Employer:
![image](https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/assets/90120572/bfd16db3-468a-4805-86bf-a5f82ebe9fe5)

Sign In Employer:
![image](https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/assets/90120572/069ff1e0-9738-4b49-9884-173990b31491)


Mobile:
![image](https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/assets/90120572/26a6ca13-9d44-428c-a81d-c046eb5e60ad)

vladjsp commented:
Nice job. But  I don't understand why you need to add custom styles to Input component. I completely comment out the styles (see picture) and remove the `className={styles.input}` from the SignInForm and it works well. It's good idea to keep functionalities for passing custom styles but in this situation it looks unnecessary.
The only thing I changed was to update the colors in the styles of the Input component as follows:
```
--white-color -> --white
--error-main -> --error-primary
--input-main -> --botticelli
--hover -> --st-tropaz
--bt-blue-gray -> --nepal
--light-background-color -> --background-primary
```
On the left side are the old colors, and on the right side are the new ones that are present in the project. With these changes we can get :hover, :focus and :error states as shown on picture.
![sign-in](https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/assets/118971905/57022c46-9040-4a17-8ccc-3a2b3c275a1d)

vladjsp commented:
And this happens on sign-up with error
![sign-up](https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/assets/118971905/d3bfcc8c-6728-4c71-b9d2-52d41db755df)

JosuerBague commented:
> Nice job. But I don't understand why you need to add custom styles to Input component. I completely comment out the styles (see picture) and remove the `className={styles.input}` from the SignInForm and it works well. It's good idea to keep functionalities for passing custom styles but in this situation it looks unnecessary. The only thing I changed was to update the colors in the styles of the Input component as follows:
> 
> ```
> --white-color -> --white
> --error-main -> --error-primary
> --input-main -> --botticelli
> --hover -> --st-tropaz
> --bt-blue-gray -> --nepal
> --light-background-color -> --background-primary
> ```
> 
> On the left side are the old colors, and on the right side are the new ones that are present in the project. With these changes we can get :hover, :focus and :error states as shown on picture. ![sign-in](https://user-images.githubusercontent.com/118971905/264127513-57022c46-9040-4a17-8ccc-3a2b3c275a1d.png)

I already removed these styles. I was using the component incorrectly, hence the confusion with the styles.

JosuerBague committed:
bt-115: + Add role to UserSignUpRequestDto and ValidationSchema

JosuerBague committed:
bt-115: * Minor changes to radio components

JosuerBague committed:
bt-115: * Minor changes to sign-in form

JosuerBague committed:
bt-115: + Add role radio group to sign-up page

JosuerBague committed:
Merge branch 'development' into task/bt-115-create-sign-in-sign-up-pages

Monyster commented on frontend/src/bundles/auth/components/sign-in-form/sign-in-form.tsx:
```diff
@@ -1,16 +1,136 @@
+        [handleSubmit, onSubmit],
+    );
+
+    const sellingPointText = isTalent
```
I think sellingPointText is useless. 
Because when someone sign-in, we don't know whether it's an employer or a talent.
It may be used to customize the sign-up page.

nikita-remeslov replied:
@Monyster agree

nikita-remeslov replied:
Let's make one page for SignIn, and change this text automatically each 3-5 seconds

Just add one text for now. After we will make some animation - https://codepen.io/alvarotrigo/pen/PoKMyWE

JosuerBague replied:
removed

Monyster commented on frontend/src/bundles/auth/components/sign-up-form/constants/constants.ts:
```diff
@@ -3,6 +3,7 @@ import { type UserSignUpRequestDto } from '~/bundles/users/users.js';
 const DEFAULT_SIGN_UP_PAYLOAD: UserSignUpRequestDto = {
     email: '',
     password: '',
+    role: 'talent',
```
Merge development and use shared enum

Monyster commented on frontend/src/bundles/auth/components/sign-up-form/sign-up-form.tsx:
```diff
@@ -1,17 +1,60 @@
+import styles from './styles.module.scss';
 
 type Properties = {
+    isTalent: boolean;
```
```suggestion

```
Create useState for it inside the form

Monyster commented on frontend/src/bundles/auth/components/sign-up-form/sign-up-form.tsx:
```diff
@@ -24,26 +67,86 @@ const SignUpForm: React.FC<Properties> = ({ onSubmit }) => {
         [handleSubmit, onSubmit],
     );
 
+    const sellingPointText = isTalent
```
Create useState for this logic

Monyster commented on frontend/src/bundles/auth/pages/auth.tsx:
```diff
@@ -31,10 +31,14 @@ const Auth: React.FC = () => {
             case AppRoute.SIGN_IN: {
-                return <SignInForm onSubmit={handleSignInSubmit} />;
+                return (
+                    <SignInForm isTalent={true} onSubmit={handleSignInSubmit} />
```
```suggestion
                    <SignInForm onSubmit={handleSignInSubmit} />
```
I think when we go to the signin form, we don't know who our client is

Monyster commented on frontend/src/bundles/auth/pages/auth.tsx:
```diff
@@ -31,10 +31,14 @@ const Auth: React.FC = () => {
             case AppRoute.SIGN_UP: {
-                return <SignUpForm onSubmit={handleSignUpSubmit} />;
+                return (
+                    <SignUpForm isTalent={true} onSubmit={handleSignUpSubmit} />
```
```suggestion
                    <SignUpForm onSubmit={handleSignUpSubmit} />
```
I think we need to know who wants to signup inside the form using useState, for example

Monyster commented:
@JosuerBague Something changed in the design
![image](https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/assets/42421305/1a4a362b-4693-4ba3-8640-7cfbcfd826fc)

ptarho commented:
Also there should be a text in bottom left corner and sign-in action is not implemented

nikita-remeslov commented on frontend/src/bundles/auth/components/google-button/google-button.tsx:
we don't need google button

nikita-remeslov commented on frontend/src/assets/img/google-icon.svg:
the same

nikita-remeslov commented on frontend/src/bundles/auth/components/sign-up-form/sign-up-form.tsx:
```diff
@@ -1,17 +1,60 @@
+    },
+];
+
+const renderRadio = ({
```
1 component === 1 file

nikita-remeslov reviewed (changes requested):
![image](https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/assets/78347072/d16326a8-952b-4e0d-8649-11393540507f)
not all elements have the same indentation (checkbox)

styles duplicated

you can make nested routes

nikita-remeslov commented:
> @JosuerBague Something changed in the design ![image](https://user-images.githubusercontent.com/42421305/264345716-1a4a362b-4693-4ba3-8640-7cfbcfd826fc.png)

Icon we will update in another task

JosuerBague committed:
Merge branch 'development' into task/bt-115-create-sign-in-sign-up-pages

JosuerBague committed:
bt-115: * Removed google button

JosuerBague committed:
bt-115: * Minor changes

JosuerBague committed:
Merge branch 'development' into task/bt-115-create-sign-in-sign-up-pages

JosuerBague committed:
bt-115: * Fix merge conflicts

JosuerBague committed:
bt-115: + Add Auth Layout component

JosuerBague committed:
Merge branch 'development' into task/bt-115-create-sign-in-sign-up-pages

JosuerBague committed:
bt-115: + Add mock alert to sign-in

JosuerBague committed:
Merge branch 'development' into task/bt-115-create-sign-in-sign-up-pages

JosuerBague committed:
bt-115: * Minor changes

nikita-remeslov reviewed (changes requested):
![image](https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/assets/78347072/5addbe71-a21d-49fa-b3a0-9fc2ae542cfc)

nikita-remeslov commented on frontend/src/bundles/auth/components/sign-up-form/sign-up-form.tsx:
```diff
@@ -24,26 +55,82 @@ const SignUpForm: React.FC<Properties> = ({ onSubmit }) => {
+        fieldState: ControllerFieldState;
+        formState: UseFormStateReturn<UserSignUpRequestDto>;
+    }): React.ReactElement {
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
```
```suggestion
```

JosuerBague replied:
If I don't destructure out ref:
![image](https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/assets/90120572/a1cc0841-f8fc-4c76-8818-d9cc338b4e9c)


I get this error:
![image](https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/assets/90120572/b661780d-49cb-4c70-b0c0-6197232bd3a7)

nikita-remeslov replied:
why do we use Controler? Can we validate the form without this component?

JosuerBague replied:
I am currently researching this, because all examples I see use a variant of this:
```
<FormControl>
         <TextField label="Email *" {...register('email', { required: true})} name='email'/>
</FormControl>
```

And it should work in theory, but for now I can't get the form to validate. Will come back with more info on this.

JosuerBague replied:
Bugfix bt-10 has a working radio-group that removes the need to use `Controller`.

nikita-remeslov commented on frontend/src/bundles/auth/components/sign-up-form/sign-up-form.tsx:
```diff
@@ -1,16 +1,47 @@
+    },
+    {
+        value: UserRole.TALENT,
+        // eslint-disable-next-line @typescript-eslint/quotes
```
```suggestion
```

JosuerBague replied:
Github Action CI linter throws an error if I use escaped characters.

JosuerBague replied:
![Uploading image.png…]()

nikita-remeslov commented on frontend/src/bundles/auth/components/sign-up-form/sign-up-form.tsx:
```diff
@@ -1,16 +1,47 @@
+    {
+        value: UserRole.TALENT,
+        // eslint-disable-next-line @typescript-eslint/quotes
+        label: "I'm looking for a job",
```
```suggestion
        label: 'I\'m looking for a job',
```

JosuerBague replied:
Github Action CI linter throws an error if I use escaped characters.

nikita-remeslov commented on frontend/src/bundles/auth/components/sign-up-form/sign-up-form.tsx:
```diff
@@ -1,16 +1,47 @@
+const options = [
+    {
+        value: UserRole.EMPLOYER,
+        // eslint-disable-next-line @typescript-eslint/quotes
```
```suggestion
```

nikita-remeslov commented on frontend/package.json:
```diff
@@ -9,7 +9,7 @@
     "scripts": {
         "lint:css": "npx stylelint \"src/**/*.scss\"",
-        "lint:js": "npx eslint \"src/**/*.{ts,tsx}\"",
+        "lint:js": "npx eslint \"src/**/*.{ts,tsx}\" --fix",
```
```suggestion
        "lint:js": "npx eslint \"src/**/*.{ts,tsx}\"",
```

nikita-remeslov replied:
We do not need to change this command
If you want to run fix locally you can pass params to the command

`npm run lint:js -- --fix`

Even if we need to add such changes, it should be discussed with the team

JosuerBague committed:
bt-115: * Removed es-lint disables

JosuerBague committed:
bt-115: - Removed changes concerning sign in API

JosuerBague committed:
Merge branch 'development' into task/bt-115-create-sign-in-sign-up-pages

JosuerBague committed:
bt-115: * Minor changes

JosuerBague committed:
Merge branch 'fix/bt-10-add-radio-component' into task/bt-115-create-sign-in-sign-up-pages

nikita-remeslov commented on frontend/src/bundles/auth/components/sign-up-form/styles.module.scss:
```diff
@@ -0,0 +1,95 @@
+.form {
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    align-items: center;
+    margin-top: calc(min(10vh, 228px));
+    margin-bottom: calc(min(10vh, 228px));
+}
+
+.header {
+    width: 100%;
+    max-width: 420px;
+    font-weight: var(--font-weight-bold);
+    font-size: var(--font-size-h4);
+    font-family: var(--font-family);
+    line-height: var(--line-height-main);
+    text-align: center;
+}
+
+.email-sign {
+    margin-top: 27px;
+    margin-bottom: 23px;
+    color: var(--gray-1);
+    font-size: var(--font-size-caption);
+    font-family: var(--font-family);
+}
+
+.input-container {
+    width: 100%;
+
+    &.email {
+        margin-bottom: 25px;
+    }
+
+    &.password {
+        margin-bottom: 8px;
+    }
+}
```
why do we have so many duplicate styles?

JosuerBague replied:
consolidated into form-styles

nikita-remeslov commented on frontend/src/bundles/auth/components/sign-in-form/styles.module.scss:
```diff
@@ -0,0 +1,116 @@
+.form {
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    align-items: center;
+    margin-top: calc(min(10vh, 228px));
+    margin-bottom: calc(min(10vh, 228px));
+}
+
+.header {
+    width: 100%;
+    max-width: 420px;
+    font-weight: var(--font-weight-bold);
+    font-size: var(--font-size-h4);
+    font-family: var(--font-family);
+    line-height: var(--line-height-main);
+    text-align: center;
+}
+
+.email-sign {
+    margin-top: 27px;
+    margin-bottom: 23px;
+    color: var(--gray-1);
+    font-size: var(--font-size-caption);
+    font-family: var(--font-family);
+}
+
+.input-container {
+    width: 100%;
+    max-width: 420px;
+
+    &.email {
+        margin-bottom: 25px;
+    }
+
+    &.password {
+        margin-bottom: 8px;
+    }
+}
```
why do we have so many duplicate styles?

JosuerBague committed:
Merge branch 'development' into task/bt-115-create-sign-in-sign-up-pages

JosuerBague committed:
bt-115: + Refactored talent radio group to not use Controller

JosuerBague committed:
bt-115: * Moved auth forms to auth-forms dir

JosuerBague committed:
bt-115: * Minor style change for footer

JosuerBague committed:
Merge branch 'development' into task/bt-115-create-sign-in-sign-up-pages

JosuerBague committed:
bt-115: * Moved form styles to auth dir
